### PR TITLE
Add irap_fedora container

### PR DESCRIPTION
**Pull request by pmb59**
This builds a new container with irap_fedora (Integrated RNA-Seq Analysis Pipeline) from Docker hub and adds new necessary functionality (time).

### CODE REVIEW

File: irap-fedora/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM nunofonseca/irap_fedora:latest
```
Comment (pcm32) [inline irap-fedora/Dockerfile:1]: Please use the exact tag that we were using, I had to go through many versions to sort of match the versions of tools that we were using on Noah.

Comment (pcm32) [reply irap-fedora/Dockerfile:1]: v1.0.3

File: irap-fedora/image_tag [hunk 2]
```diff
@@ -0,0 +1 @@
+irap-fedora:0.0.1
```
Comment (pcm32) [inline irap-fedora/image_tag:1]: I would suggest
```suggestion
irap-fedora:v1.0.3+atlas0
```

File: irap-fedora/Dockerfile (hunk 1 above)
Comment (pcm32) [reply irap-fedora/Dockerfile:1]: (or at least to be contemporary to the current noah installation)

**Comment by irisdianauy** [comment]
Closing this PR and starting a fresh PR to see if CI runner is triggered using the `on: pull_request` info.

**Comment by irisdianauy** [comment]
Sorry, leaving this open for now and will simply start a new PR.

File: .github/workflows/irap-fedora-ci-runner.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  #   - name: Build image
+  #     id: build-image
+  #     uses: redhat-actions/buildah-build@v2
+  #     with:
+  #       image: ${{ env.IMAGE_NAME }}
+  #       tags: ${{ steps.gettag.outputs.tag }}
+  #       containerfiles: |
+  #         ./irap-fedora/Dockerfile
+    
+  #   - name: Push to Quay
```
Comment (pinin4fjords) [inline .github/workflows/irap-fedora-ci-runner.yaml:58]: When you re-enable this, should it have a filter so it only happens with master commits? See e.g. https://github.com/ebi-gene-expression-group/atlas-containers/blob/40b787e88c7e4bb7a2407f05ae31ba3f849d3016/.github/workflows/conda-templated.yaml#L88

Comment (irisdianauy) [reply .github/workflows/irap-fedora-ci-runner.yaml:58]: Thanks for this, I'll add that filter!

File: irap-fedora/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,7 @@
+FROM nunofonseca/irap_fedora:v1.0.3
+
+# Install time
+RUN yum -y update && yum -y install time
+
+# Add libtbb.so.2 to LD_LIBRARY_PATH
+RUN export LD_LIBRARY_PATH=/opt/irap/bin/salmon/lib:$LD_LIBRARY_PATH
```
Comment (pcm32) [inline irap-fedora/Dockerfile:7]: to set env variables you should use the ENV directive, not RUN.

File: irap-fedora/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FROM nunofonseca/irap_fedora:v1.0.3
+
+# Install time
+RUN yum -y update && yum -y install time
+
+# Add libtbb.so.2 and the appropriate libm.so.6 to LD_LIBRARY_PATH
+ENV LD_LIBRARY_PATH=/opt/irap/bin/salmon/lib/libtbb.so.2:/usr/lib64
```
Comment (pcm32) [inline irap-fedora/Dockerfile:7]:
```suggestion
ENV LD_LIBRARY_PATH=/opt/irap/bin/salmon/lib:/usr/lib64:$LD_LIBRARY_PATH
```
unless that before it was just /usr/lib64 only. Still you probably want to cut shot to salmon/lib.

Comment (irisdianauy) [reply irap-fedora/Dockerfile]: This is the first occurrence of LD_LIBRARY_PATH in the container.  There is no LD_LIBRARY_PATH set in the container until the sourcing of the script that sets up the irap environment (which expects and appends LD_LIBRARY_PATH), so I didn't include `:$LD_LIBRARY_PATH` in this line.  I've also added /usr/lib64 so the container can use the correct `libm.so.6`.  This is to resolve a complaint from bedtools stemming from the use of the incompatible `/opt/irap/bin/salmon/liblibm.so.6` (also the reason why I specified the salmon/lib/libtbb.so.2, but the change in preference in 7988421a0ff55ba5dfe001f8d3d2ec7b75de358e resolves that cleanly).

File: .github/workflows/irap-fedora-ci-runner.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  pull_request:
+    paths:
+      - "irap-fedora/**"
+
+jobs:          
+  deploy:
+    name: Build and deploy image to quay.io/ebigxa
+    runs-on: ubuntu-20.04
+    env:
+      IMAGE_NAME: irap_fedora
```
Comment (pcm32) [inline .github/workflows/irap-fedora-ci-runner.yaml:18]: The IMAGE_NAME should come as well from the tag file, as the tag comes. I'm sure the CI for conda templates has this already and you should be able to copy paste.

File: .github/workflows/irap-fedora-ci-runner.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        tags: ${{ steps.gettag.outputs.tag }}
+        containerfiles: |
+          ./irap-fedora/Dockerfile
+    
+    - name: Push to Quay
+      if: ${{ github.ref == 'refs/heads/master' }}
+      id: push-to-quay
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: ${{ steps.build-image.outputs.image }}
```
Comment (pcm32) [inline .github/workflows/irap-fedora-ci-runner.yaml:41]: if the quay.io/ebigxa/irap_fedora doesn't exist yet, you might need to create it there before this actually works (see general README on this repo).

**Review by pcm32 — COMMENTED** [review]
LGTM in general, please address the comments I mention though. Good work!

File: .github/workflows/irap-fedora-ci-runner.yaml (hunk 7 above)
Comment (irisdianauy) [reply .github/workflows/irap-fedora-ci-runner.yaml]: Done, with appropriate write access.

File: .github/workflows/irap-fedora-ci-runner.yaml (hunk 6 above)
Comment (irisdianauy) [reply .github/workflows/irap-fedora-ci-runner.yaml]: Done in 69f474a98323110c9e6958f04634890e27de9c2b

**Review by pcm32 — APPROVED** [review]
LGTM, thanks for addressing my comments.

**Comment by pcm32** [comment]
This could probably be generalised later on to a simple dockerfile template (for maybe directories starting with simple_dockerfile_<name>), then they all reuse the same CI. But no need now.

**Comment by irisdianauy** [comment]
Thanks @pinin4fjords and @pcm32!

**Comment by pcm32** [comment]
There is a Jenkins job to transform this now to singularity on the cluster, si that it can be used by ISL

**Pull request merged**